app/controllers/users_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  before_action :check_already_signed_up, only: :sign_up
+
+  VALID_USER_TYPES = %w(driver passenger).freeze
+
+  def sign_in
+    user_params = sign_in_params
+
+    user = User.find_by(email: user_params[:email])
+    raise Exceptions::BadRequest, '아이디와 비밀번호를 확인해주세요' unless user&.authenticate(user_params[:password])
+
+    log_in_user(user)
+
+    json_success({ accessToken: user.token })
+  rescue ActionController::ParameterMissing
+    render json: { message: '아이디와 비밀번호를 확인해주세요' }, status: Exceptions::BadRequest.http_status
+  end
+
+  def sign_up
+    user = CreateUserService.new(sign_up_params).run!
+
+    json_create_success(UserSerializer.new(user).as_json)
+  rescue ActionController::ParameterMissing
+    message = if !valid_email?(params[:email])
+                '올바른 이메일을 입력해주세요'
+              elsif params[:password].blank?
+                '올바른 비밀번호를 입력해주세요'
+              elsif !valid_user_type?(params[:userType])
+                '올바른 사용자 타입을 입력해주세요'
+              end
+    raise Exceptions::BadRequest, message
+  end
+
+  private
+
+  def sign_in_params
+    params.require(%i(email password))
+    params.permit(%i(email password))
+  end
+
+  def sign_up_params
+    params.require(%i(email password userType))
+
+    raise Exceptions::BadRequest, '올바른 이메일을 입력해주세요' unless valid_email?(params[:email])
+    raise Exceptions::BadRequest, '올바른 사용자 타입을 입력해주세요' unless valid_user_type?(params[:userType])
+
+    params.permit(%i(email password userType))
+  end
+
+  def check_already_signed_up
+    raise Exceptions::Conflict, '이미 가입된 이메일입니다' if User.find_by(email: params[:email]).present?
```

Comment:
값에 대한 존재 유무를 검색할땐 exists? 메소드를 사용해주세요.
그리고 이 validation은 User 모델에 있는게 좋을 것 같아요.
https://guides.rubyonrails.org/active_record_validations.html#uniqueness
```suggestion
    raise Exceptions::Conflict, '이미 가입된 이메일입니다' if User.exists?(email: params[:email])
```

Reply:
모델에 uniqueness validation 추가하고, 생성 시 이 validation에 걸리면 controller에서 에러 raise하도록 [291b4ca](https://github.com/travelingaries/taxi-dispatch-backend-api/pull/1/commits/291b4cabe1e033d151f7dabfe86ff0be232012ba)에서 플로우 수정했습니다 🙆‍♂️ 
혹시 생각하고 있으셨던 로직이 요게 아니라면 알려주세요